### DISCUSSION TITLE
change DeviceName

### DISCUSSION THREAD

**Discussion by kareemkhalil** [arendst/Tasmota#15824] [Q&A] [answered]
Hello Dears, 
i am starting to compile tasmota and i want to change the default DeviceName, 
and if their is some documentation for the variables that could be changed it will be grate.

**Comment by sfromis** [comment, marked as answer]
Looks like you can do a `#define DEVICE_NAME "MyDeviceName"`. 

That name was not included in the usual place where you can find "most" available build time default variables, [my_user_config.h](https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h), for deciding what to put into `user_config_override.h`

**Reply by kareemkhalil** [reply]
thank you i will give it a try, but i was wondering if i want to have a unique name such as today date or like Tasmota  do for mqtt use the last 6 digits in mac address. is it doable 

i am sorry maybe my question is dummy but i am new for this kind of developing :)

**Reply by kareemkhalil** [reply]
can i use something like this? 

"DVES_%06X"

**Reply by sfromis** [reply]
The device name is intended as a user-oriented "friendly" text including blanks, mixed case and so on, not as a cryptic id (which is more suitable in a topic or hostname). Therefore I do not think (not tried it) that the same sort of replacements are available in the device name, as  using things like MAC or date was just not seen as relevant.
